### PR TITLE
Fix jena dependency issue (#1622)

### LINKED ISSUE #1622: Dependency issue with jena coming with new metafix  "Skos-lookup"
The [Skos-lookup implementation introduces a dependency to jena](https://github.com/metafacture/metafacture-fix/commit/b55bbbf508b2cd484abfa4221649589c06c9db7d#diff-67162ab57d816849d43c537ebdab31a8744ac824f4f512d0462a78d072937a4a) which breaks the use of jena in play app
=> compiling error:
> web/app/controllers/resources/RdfConverter.java:12: package com.hp.hpl.jena.rdf.model does not exist

### PR DESCRIPTION
**Pull request by dr0i**
With https://github.com/metafacture/metafacture-fix/commit/b55bbbf508b2cd484abfa4221649589c06c9db7d#diff-67162ab57d816849d43c537ebdab31a8744ac824f4f512d0462a78d072937a4a a new jena dependency is transitively introduced via metafix.

- update "jena-arq" from "2.10.1" to "3.17.0"
- exclude "commons-codec" from "jsonld-java" dependency
- refactor classes
- remove unnecassary dependencies in buid.sbt (jsonld and jena)
- 
Resolves #1622

### CODE REVIEW

File: src/main/java/de/hbz/lobid/helper/CreateWikidataNwbibMaps.java [hunk 1]
```diff
@@ -9,18 +9,18 @@
...
 
-import com.hp.hpl.jena.rdf.model.Model;
-import com.hp.hpl.jena.rdf.model.ModelFactory;
-import com.hp.hpl.jena.rdf.model.Property;
-import com.hp.hpl.jena.rdf.model.ResIterator;
-import com.hp.hpl.jena.rdf.model.Resource;
 
 /**
  * Creates nwbib maps used by morph.
- *
```
Comment (fsteeg) [inline src/main/java/de/hbz/lobid/helper/CreateWikidataNwbibMaps.java:23]: There seem to be a lot of these blankspace / formatting changes.

File: src/main/java/org/lobid/resources/Triples2RdfModel.java [hunk 2]
```diff
@@ -92,31 +92,30 @@ public void setPropertyIdentifyingTheNodeForInferencing(String property) {
...
+            }
+        }
 	}
 
 	private boolean validTriple(String triple, Model model) {
 		try {
 			model.read(new StringReader(triple), "test:uri" + count++, serialization);
 		} catch (Exception e) {
-			LOG.info("... ignore broken triple: " + triple);
+			LOG.info("... ignore broken triple");
```
Comment (fsteeg) [inline src/main/java/org/lobid/resources/Triples2RdfModel.java:118]: Is this intentional? No details available here?

File: web/app/controllers/resources/RdfConverter.java [hunk 3]
```diff
@@ -47,9 +45,11 @@ public String getName() {
 	 */
 	public static String toRdf(final String jsonLd, final RdfFormat format) {
 		try {
-			final Object jsonObject = JsonUtils.fromString(jsonLd);
-			final JenaTripleCallback callback = new JenaTripleCallback();
-			final Model model = (Model) JsonLdProcessor.toRDF(jsonObject, callback);
+			//convert json-ld string into InputStream as is required by the read() function.
+			InputStream targetStream = new ByteArrayInputStream(jsonLd.getBytes());
+			Model model = ModelFactory.createDefaultModel() ;
```
Comment (fsteeg) [inline web/app/controllers/resources/RdfConverter.java:50]: Other variables here are declared `final`, not sure if it's an intended change.

**Review by fsteeg — APPROVED** [review]
Quick review to unblock since I'm now off for a week. Commented some minor things I noticed inline.

File: src/main/java/de/hbz/lobid/helper/CreateWikidataNwbibMaps.java (hunk 1 above)
Comment (dr0i) [reply src/main/java/de/hbz/lobid/helper/CreateWikidataNwbibMaps.java:23]: Yes, they were formely unintendedly  introduced and now fixed by my IDE automatically.

File: src/main/java/org/lobid/resources/Triples2RdfModel.java (hunk 2 above)
Comment (dr0i) [reply src/main/java/org/lobid/resources/Triples2RdfModel.java:118]: An ErrorHandler changed it's behaviour in newer jena lib: ErrorHandlerFactory writes ERROR messages which hereby will be infoed as to be ignored (tried to configure the ErrorHandler to not be strict but didn't achieve that. Would need some more refactoring. )

File: web/app/controllers/resources/RdfConverter.java (hunk 3 above)
Comment (dr0i) [reply web/app/controllers/resources/RdfConverter.java]: Not intended. IDE didn't take care and I forgot about it. Fixed. Thx.

**Comment by dr0i** [comment]
Thx for the fast review @fsteeg.

**Pull request merged**